### ISSUE TITLE
Questions regarding the data preprocessing hyperparameters of the RealEstate10K dataset

### ISSUE THREAD

**Issue by hubert0527** [facebookresearch/synsin#6]
It seems like the evaluation of SynSin on RealEstate10K is performed on 256x256 images. However, I couldn't find some of the data preprocessing hyperparameters in the `REALESTATE.md` for results reproduction, such as (i) what resolution of the raw video did the authors download and work with, (ii) how was the 256x256 patch extracted from the video frames (e.g., center crop? or center crop after resizing?), and (iii) how did the author choose the video frames to evaluate within each video clip (each video clip has a huge set of labeled frames in RealEstate10K)?

Could the authors make a clarification on these hyperparameters?
Sincerely thanks!

**Comment by oawiles** [comment]
Hi! 
(iii) We obtained pairs by using the code from the stereo magnification paper originally. 
We randomly chose the images by running their [evaluation script](https://github.com/google/stereo-magnification/blob/master/test.py) (which randomly selects pairs of images). 

The stereo magnification work operates and evaluates on the full image. (ii) So we did this as well -- modifying the intrinsics in order to account for the change in x/y scale -- we ran on square images, but our method could be easily modified to have kept the original aspect ratio. 
When downloading videos (i) -- I believe we downloaded the videos at the original size and then resized them to the 256x256 size (I do not remember whether on disk they were their original size, or resized). 

Hope that answeres your questions!

**Comment by hubert0527** [comment]
Hi thanks for the reply!
The stereo magnification was evaluated at a 576x1024 resolution, which is not a square image. 

> The stereo magnification work operates and evaluates on the full image. (ii) So we did this as well...
So you evaluated SynSin at 256x455 (resized from 576x1024) resolution?
Also, I found the evaluation script provided in this repository resizes each input to 256x256:
https://github.com/facebookresearch/synsin/blob/82ff948f91a779188c467922c8f5144018b40ac8/evaluation/eval_realestate.py#L43
I'm not sure if this script is corresponding to the result numbers reported in the paper or not.

And for the training, do you mean you directly resize the frames (resolution 576x1024) into 256x256 (disregards the aspect ratio)?

**Comment by oawiles** [comment]
We re-evaluated the stereo-magnification numbers within our pipeline (sorry that that was unclear). 

Yes.

**Comment by hubert0527** [comment]
Oh, so you train and evaluate both Stereo Magnification and SynSin with full images that are directly resized into 256x256? And when you mention the aspect ratio, you mean you modified the intrinsic matrix to account for the change in FoV?

> we ran on square images, but our method could be easily modified to have kept the original aspect ratio.

I had a hard time trying to keep the original aspect ratio of image spatial size (576:1024) since the depth estimation module in SynSin adopts a UNet architecture, which is hard to deal with image resolutions that are not a multiple of 256.

**Comment by oawiles** [comment]
We don't retrain Stereo Mag on 256x256. We use their pretrained model, run it on the test set to get their predicted images at the original resolution (1024x576). We then resize this to 256x256 when evaluating using our evaluation pipeline (our implementations of SSIM/PSNR/PercSim). So it is only when computing the metrics that the image size is changed for StereoMag.

For our training pipeline: yes we modified the intrinsic matrix to into account the the change in FoV.

Yes, keeping the original aspect ratio was a pain, and to maintain consistency with the other datasets, we resized the RealEstate10K images to 256x256 when training/testing. You could try an hourglass model instead.

**Comment by hubert0527** [comment]
OK! Sincerely thanks for the clarification!

**Comment by MotorCityCobra** [comment]
hubert0527,
Hi. I'm not having much luck with the contributors of the repo in using the RealEstate10K dataset. Could you please tell me, how did you get the frames extracted? In other words, how did you get the dataset of just the youtube links and tensors converted to frames for the model to use? Did you start with the `Interactive Demo` notebook or is there something somewhere else in the repo that extracts the frames? I assume I shouldn't have to download each with youtube-dl and extract the frames with ffmpeg?

**Comment by phongnhhn92** [comment]
> hubert0527,
> Hi. I'm not having much luck with the contributors of the repo with figuring out how to use the RealEstate10K dataset. Could you please tell me, how did you get the frames extracted? In other words, how did you get the dataset of just the youtube links and tensors converted to frames for the model to use? Did you start with the `Interactive Demo` notebook or is there something somewhere else in the repo that extracts the frames? I assume I shouldn't have to download each with youtube-dl and extract the frames with ffmpeg?

I am working on a script to download the RealEstate dataset and yes, you are right my friend. You have to download each youtube video seperately and extract frame by frame ^^ You dont need any code from this repo to do that. Pure python programming can help you, please stop annoying the contributors and other on something in a very basic level like this.

**Comment by MotorCityCobra** [comment]
>^^ You dont need any code from this repo to do that. Pure python programming can help you, please stop annoying the contributors and other on something in a very basic level like this.

What are you naming each frame? Are you naming each png after its timestamp?

**Comment by hubert0527** [comment]
Hi @MotorCityCobra 
Sorry for the late reply, I have limited access to the internet these days. 
I downloaded videos with youtube-dl and extract them with some python libraries (can't recall which one I used). A video will be decomposed into N images (each of them is called a frame). Then things become complicated when it comes to which frames should be chosen for the training and evaluation. Usually, there will be some rules to constraint the difficulty of the novel-view camera pose (such as not having the time steps of the source view and novel view too far away) while the two views still have observable differences. Not sure if this answers your question.
